src/docfx/build/page/Markup.cs

```diff
@@ -104,9 +85,29 @@ public static (MarkdownDocument ast, MarkupResult result) Parse(string content)
             }
         }
 
-        private static MarkdownPipeline CreateConceptualMarkdownPipeline()
+        private static MarkdownPipeline GetPipeline(MarkdownPipelineType pipelineType, string locale)
+        {
+            return s_pipelineMapping.GetOrAdd(pipelineType, key => new Lazy<MarkdownPipeline>(() =>
```

Comment:
This is wrong if we build multiple locales in a single process
Keep the existing pipeline map. put `locale` into the thread static `Status` object.